src/jitlayers.cpp

```diff
@@ -0,0 +1,392 @@
+// This file is part of Julia.
+// Parts of this file are copied from LLVM, under the UIUC license.
```

Comment:
ianal, but i this doesn't sound like a valid license, and it doesn't sound like a good situation for this file. i would prefer that any content you add that requires a special license to be placed in a separate file.

fwiw, i used Kaleidoscope tutorial as the reference for learning how to write an OrcJIT, but the actual implementation of JuliaOJIT I believe should be considered as MIT licensed like the rest of codegen